src/client/pages/Welcome.stories.tsx

```diff
@@ -2,11 +2,19 @@ import React from 'react';
 import { Meta } from '@storybook/react';
 
 import { Welcome } from './Welcome';
+import { MemoryRouter } from 'react-router-dom';
 
 export default {
   title: 'Pages/Welcome',
   component: Welcome,
   parameters: { layout: 'fullscreen' },
+  decorators: [
+    (Story) => (
+      <MemoryRouter>
+        <Story />
+      </MemoryRouter>
```

Comment:
I'm not sure I understand what's going on here, why do we need the `MemoryRouter`?

Reply:
Ah wait, you need it in the test/story to render the component, as `location` would not exist otherwise.

Reply:
I wonder if it's worth moving the `useLocation` from the `Welcome.tsx` to `WelcomePage.tsx` and passing in search as a `prop`, this would mean you wouldn't need the dependency on `react-router-dom` here, and the `Welcome` component could be rendered independently.

Reply:
you could either pass in `search` as the prop, or go a level lower and pass in `viewId` as a prop to the welcome component as an optional prop.

Reply:
I like the idea of passing down `viewId`. To me passing `viewId` as a prop shows intent more clearly than passing  `search` which could be used for anything. keeps the URL param fetching logic all in one place too